packages/@aws-cdk/aws-eks/lib/cluster-resource-provider.ts

```diff
@@ -66,6 +73,9 @@ export class ClusterResourceProvider extends NestedStack {
   private constructor(scope: Construct, id: string, props: ClusterResourceProviderProps) {
     super(scope as CoreConstruct, id);
 
+    // Allow user to override the layer. Layer must contain `proxy-agent` node_module which is required to proxy AWS SDK requests.
+    const proxyAgentLayer = props.proxyAgentLayer ? props.proxyAgentLayer : new NodeProxyAgentLayer(this, 'NodeProxyAgentLayer');
```

Comment:
I dont think this should be configurable. The lambda function code now **must** have this layer in order to work, so we should simply be adding it in addition to the externally provided layer.